legacy/reqcounter/reqcounter_test.go

```diff
@@ -124,3 +126,57 @@ func TestRequestCounterIncrement(t *testing.T) {
 	// Ensure the request counter was incremented.
 	require.EqualValues(t, &expected, &requestCounter, "The request counter failed to increment its request field.")
 }
+
+func TestLog(t *testing.T) {
```

Comment:
As we discussed, this would more bucket under an integration test so it's important to test the various things we want to happen

1. Initialize the requestCounters
2. "Advance time"
3. monitor.Increment a bunch of times
4. repeat 2,3 a bunch of times
5. Check the log statements of each counter (length, order and specific values)
(we don't care about relative ordering across counters but we do want to check the order in which logs are generated by a single counter)

Reply:
Orthogonally, let's also reconsider if we are actually getting the required signal from this test if we are mocking/synchronizing a bunch of unexpected behavior just to make the integration test work.

It's okay if we can cover the individual functionality by way of focused unit tests.